### PR TITLE
doc: Added suit and smp doc

### PR DESCRIPTION
**Pull request by FrancescoSer**
Added suit and smp doc.

### CODE REVIEW

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+==================
+
+SUIT defines a *component* as an updatable logical block, which can be firmware, software, configuration, or data.
+This is similar to how MCUmgr defines an *image*.
+Updates are delivered to the device as a SUIT *envelope,* containing SUIT manifests and, optionally, integrated payloads.
+The relationships between components are maintained through SUIT manifests.
+
+The SUIT envelope serves as the main metadata source for the payload.
+Adding metadata inside the image can lead to synchronization issues, as the lifecycles of the envelope and image may diverge.
+Therefore, manage metadata at the envelope level.
```
Comment (SylwesterKonczyk) [inline doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst:37]: Lines 35 to 37... looks strange for me. If it is intended to compare SUIT to MCUboot/Zephyr, I would consider something like:

"Metadata, such a version information, expected image digest, image size is embedded into SUIT manifest"

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst [hunk 2]
```diff
@@ -0,0 +1,460 @@
...
+
+The following table summarizes key similarities and differences between MCUmgr and SUIT:
+
++-----------------------------+-------------------------+----------------------------------+
+| Feature                     | MCUmgr                  | SUIT                             |
++=============================+=========================+==================================+
+| Metadata                    | Attached to the images  | Defined by a separate SUIT       |
+|                             |                         | manifest                         |
++-----------------------------+-------------------------+----------------------------------+
+| Image Topology              | Slots (primary,         | Tree structure with manifests    |
```
Comment (SylwesterKonczyk) [inline doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst:48]: **Tree** structure with manifests grouped into a hierarchical **tree**

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst [hunk 3]
```diff
@@ -0,0 +1,460 @@
...
+
+    {
+        (str,opt)"image"    : (uint)     //SUIT - always 0 as system would be seen as "single image" system
+        (str,opt)"len"      : (uint)
+        (str)"off"          : (uint)
+        (str,opt)"sha"      : (str)
+        (str,opt)"data"     : (byte str)
+        (str,opt)"upgrade"  : (bool)
+    }
+
```
Comment (SylwesterKonczyk) [inline doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst:111]: One extra command related to SMP IMG management group shall be documanted as well:

Set state of image request

This command triggers a installation process using SUIT envelope stored in DFU partition. The CBOR data format is:

{ 
    (str)"confirm"      : (bool) 	//Must be stet to 'true'
}

**Review by SylwesterKonczyk — DISMISSED** [review]

**Review by maciejpietras — APPROVED** [review]

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst (hunk 2 above)
Comment (FrancescoSer) [reply doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst:48]: Could you describe this feature in a more verbose way? Would be helpful to edit this.

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst [hunk 4]
```diff
@@ -0,0 +1,464 @@
...
+        (str,opt)"image"    : (uint)     //SUIT - 0 - DFU partition, 1 to n - DFU cache partition 0 to n-1.
+        (str,opt)"len"      : (uint)
+        (str)"off"          : (uint)
+        (str,opt)"data"     : (byte str)
+    }
+
+Set state of images request
+---------------------------
+
+This command triggers a. installation process using the SUIT envelope stored in the DFU partition.
```
Comment (b-gent) [inline doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst:107]:
```suggestion
This command triggers an installation process using the SUIT envelope stored in the DFU partition.
```

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst [hunk 5]
```diff
@@ -0,0 +1,464 @@
...
+        (str, opt)"digest_algorithm"            : (int)
+        (str, opt)"signature_check"             : (uint)
+        (str, opt)"sequence_number"             : (uint)
+        (str, opt)"semantic_version"            : [0*5 int]
+    }
+
+where:
+
+* *class_id* - UUID representing a given manifest type
+* *vendor_id* - Vendor ID, manifest class ID was generated in the space of this value.
```
Comment (b-gent) [inline doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst:203]: either use full stops everywhere or nowhere, I suggest everywhere

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst [hunk 6]
```diff
@@ -0,0 +1,464 @@
...
+* *downgrade_prevention_policy* - Downgrade prevention policy of the given supported manifest
+* *independent_updateability_policy* - Can the given supported manifest be updated independently
```
Comment (b-gent) [inline doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst:205]:
```suggestion
* *independent_updateability_policy* - Indicates whether the given supported manifest can be updated independently.
```
(at least this is how I understand this)

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst [hunk 7]
```diff
@@ -0,0 +1,464 @@
...
+  * *MGMT_ERR_EUNKNOWN* - Some issues with NVM erase/write operations.
+
+
+Get missing image state request
+-------------------------------
+
+SUIT command sequence has the ability of conditional execution of directives, i.e., based on the digest of the installed image.
+This opens a scenario where the SUIT candidate envelope contains only SUIT manifests, and images (those required to be updated) are fetched by the device only if necessary.
+In that case, the device informs the SMP client that a specific image is required, and then the SMP client delivers the requested image in chunks.
+Due to the fact that SMP is designed in a client-server pattern and lacks server-sent notifications, implementation bases on polling.
```
Comment (b-gent) [inline doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst:306]:
```suggestion
Due to the fact that SMP is designed in a client-server pattern and lacks server-sent notifications, the implementation is based on polling.
```

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst [hunk 8]
```diff
@@ -0,0 +1,464 @@
...
+
+where:
+
+* *off* - Offset of the last successfully written byte of the candidate envelope.
+* *rc* - Return code.
+  The following values represent unrecoverable errors; once an SMP client receives any of these, it must stop the current transfer:
+
+  * *MGMT_ERR_EBADSTATE* - Possible reason - device reboot occurred in the middle of the transfer, etc.
+  * *MGMT_ERR_ENOENT* - Cache pool not found.
+    dfu_cache_partition_n not defined in device dts, or, for cache pool 0, dfu_partition not defined in dts or candidate envelope not stored in dfu partition.
```
Comment (b-gent) [inline doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst:432]:
```suggestion
    dfu_cache_partition_n not defined in device DTS, or, for cache pool 0, dfu_partition not defined in DTS or candidate envelope not stored in dfu partition.
```
assuming 'dfu partition' is its name and hence can be kept lowercase

**Review by b-gent — CHANGES_REQUESTED** [review]

**Comment by divipillai** [comment]
Could add a comment on PR https://github.com/nrfconnect/sdk-nrf/pull/18192, if changelog entry is required.

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst [hunk 9]
```diff
@@ -0,0 +1,464 @@
...
+
+SUIT and SMP protocol (including SUIT SMP extension)
+####################################################
+
+.. contents::
+   :local:
+   :depth: 2
+
+SUIT DFU offers a flexible and extensible update process.
+You can use the Zephyr management subsystem, via the SMP protocol, to manage SUIT-based images alongside existing MCUBoot images.
```
Comment (divipillai) [inline doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst:11]:
```suggestion
You can use the Zephyr management subsystem, through the SMP protocol, to manage SUIT-based images alongside existing MCUboot images.
```

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst [hunk 10]
```diff
@@ -0,0 +1,464 @@
...
++-----------------------------+-------------------------+----------------------------------+
+| State information           | Embedded within the     | Update candidate existence is    |
+|                             | image slot metadata     | represented within the SUIT      |
+|                             |                         | manifest.                        |
++-----------------------------+-------------------------+----------------------------------+
+
+SUIT in SMP image management group
+==================================
+
+The SMP protocol can provide SMP-based SUIT image management, assuming that a SUIT-based system, even a complex one, can be represented as a *single image* system.
```
Comment (annwoj) [inline doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst:67]:
```suggestion
The SMP protocol can provide SMP-based SUIT image management, assuming that even a complex SUIT-based system can be represented as a *single image* system.
```

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst [hunk 11]
```diff
@@ -0,0 +1,464 @@
...
+            (str)"version"        : (str)       //SUIT - sequence-number from root manifest.
+            (str)"hash"           : (byte str)  //SUIT - digest of root manifest.
+        }
+        ...
+    ]}
+
+Image upload request
+--------------------
+
+This command uploads an image containing a SUIT envelope (to be stored in the DFU partition), or a dump of DFU cache partition.
```
Comment (annwoj) [inline doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst:92]:
```suggestion
This command uploads an image containing a SUIT envelope to be stored in the DFU partition or a dump of the DFU cache partition.
```

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst (hunk 9 above)
Comment (annwoj) [reply doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst:11]: +Simple Management Protocol (SMP)

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst [hunk 12]
```diff
@@ -0,0 +1,464 @@
...
+Get SUIT manifests list request
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This command allows retrieving information about the roles of manifests supported by the device::
+
+    OP: MGMT_OP_READ (0), MGMT_OP_READ_RSP(1)
+
+    Command ID: 0
+
+CBOR Data of Successful Response:
```
Comment (annwoj) [inline doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst:139]:
```suggestion
CBOR data of successful response:
```

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst [hunk 13]
```diff
@@ -0,0 +1,464 @@
...
+
+* *rc* - Return code:
+
+  - **MGMT_ERR_ENOTSUP** - Unsupported command.
+  - **MGMT_ERR_EMSGSIZE** - Device was unable to encode the answer.
+
+Get SUIT manifest state request
+-------------------------------
+
+This command allows getting information about the configuration of supported manifests and selected attributes of installed manifests of a specified role.
```
Comment (annwoj) [inline doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst:164]:
```suggestion
This command allows you to get information about the configuration of supported manifests and selected attributes of installed manifests of a specified role.
```

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst [hunk 14]
```diff
@@ -0,0 +1,464 @@
...
+  Following errors are possibly recoverable:
+
+  * *MGMT_ERR_EMSGSIZE* - Device was unable to encode the answer.
+  * *MGMT_ERR_EUNKNOWN* - Some issues with NVM erase/write operations.
+
+
+Get missing image state request
+-------------------------------
+
+SUIT command sequence has the ability of conditional execution of directives, i.e., based on the digest of the installed image.
```
Comment (annwoj) [inline doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst:303]:
```suggestion
SUIT command sequence has the ability of conditional execution of directives based on the digest of the installed image.
```

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst [hunk 15]
```diff
@@ -0,0 +1,464 @@
...
+  - **MGMT_ERR_EMSGSIZE** - Device was unable to encode the answer.
+
+Get SUIT manifest state request
+-------------------------------
+
+This command allows you to get information about the configuration of supported manifests and selected attributes of installed manifests of a specified role.
+
+OP: MGMT_OP_READ (0), MGMT_OP_READ_RSP(1)
+
+Command ID: 1
```
Comment (wiba-nordic) [inline doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst:168]: These seem like they need some sort of formatting.

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst [hunk 16]
```diff
@@ -0,0 +1,460 @@
...
+--------
+
+Get SUIT manifests list request
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This command allows retrieving information about the roles of manifests supported by the device::
+
+    OP: MGMT_OP_READ (0), MGMT_OP_READ_RSP(1)
+
+    Command ID: 0
```
Comment (wiba-nordic) [inline doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst:137]: Just noticed this has the same information, but formatted differently. Either format is fine with me.

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst [hunk 17]
```diff
@@ -0,0 +1,460 @@
...
+
+Candidate envelope upload request
+---------------------------------
+
+The command delivers a SUIT envelope to the device.
+Once the upload is completed, the device validates the delivered envelope and starts SUIT processing.
+
+OP: MGMT_OP_WRITE (2), MGMT_OP_WRITE_RSP(3)
+
+Command ID: 2
```
Comment (wiba-nordic) [inline doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst:247]: This needs the format as well.

**Review by wiba-nordic — DISMISSED** [review]

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst [hunk 18]
```diff
@@ -0,0 +1,465 @@
...
+| Slot/Image Configuration    | Slots defined in        | Slot definitions can be modified |
+|                             | MCUboot; difficult to   | via the SUIT manifest, allowing  |
+|                             | change after deployment | greater flexibility              |
++-----------------------------+-------------------------+----------------------------------+
+| Installation/Boot           | Limited by static       | Customizable through SUIT        |
+| Customization               | image metadata          | manifest commands                |
++-----------------------------+-------------------------+----------------------------------+
+| State information           | Embedded within the     | Update candidate existence is    |
+|                             | image slot metadata     | represented within the SUIT      |
+|                             |                         | manifest.                        |
```
Comment (richabp) [inline doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst:61]: For consistency.
```suggestion
|                             |                         | manifest                        |
```

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst [hunk 19]
```diff
@@ -0,0 +1,465 @@
...
+
+.. note::
+   Image metadata, such as version information, is embedded within the image payload.
+
+Components in SUIT
+==================
+
+SUIT defines a *component* as an updatable logical block, which can be firmware, software, configuration, or data.
+This is similar to how MCUmgr defines an *image*.
+Updates are delivered to the device as a SUIT *envelope,* containing SUIT manifests and, optionally, integrated payloads.
```
Comment (richabp) [inline doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst:32]:
```suggestion
Updates are delivered to the device as a SUIT *envelope*, containing SUIT manifests and, optionally, integrated payloads.
```

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst [hunk 20]
```diff
@@ -0,0 +1,465 @@
...
+The relationships between components are maintained through SUIT manifests.
+
+Metadata, such a version information, expected image digest, image size is embedded into the SUIT manifest.
```
Comment (richabp) [inline doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst:35]:
```suggestion
Metadata, such as version information, expected image digest, and image size is embedded into the SUIT manifest.
```

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst [hunk 21]
```diff
@@ -0,0 +1,465 @@
...
+    }
+
+where:
+
+* *role* - Manifest role encoded as two nibbles: `<domain ID> <index>`.
+
+* *rc* - Return code:
+
+  - **MGMT_ERR_ENOTSUP** - Unsupported command.
+  - **MGMT_ERR_EMSGSIZE** - Device was unable to encode the answer.
```
Comment (richabp) [inline doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst:159]:
```suggestion
  * **MGMT_ERR_ENOTSUP** - Unsupported command.
  * **MGMT_ERR_EMSGSIZE** - Device was unable to encode the answer.
```

**Review by wiba-nordic — APPROVED** [review]

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst [hunk 22]
```diff
@@ -0,0 +1,465 @@
...
+
+  The following errors are possibly recoverable:
+
+  * *MGMT_ERR_EMSGSIZE* - The device was unable to encode the answer.
+  * *MGMT_ERR_EUNKNOWN* - Some issues with NVM erase/write operations.
+
+Cleanup request
+---------------
+
+This command erases the DFU partition and DFU Cache partitions.
```
Comment (richabp) [inline doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst:446]:
```suggestion
This command erases the DFU partition and DFU cache partitions.
```

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst [hunk 23]
```diff
@@ -0,0 +1,465 @@
...
+* *off* - Offset of the last successfully written byte of the candidate envelope.
+* *rc* - Return code.
+  The following values represent unrecoverable errors; once an SMP client receives any of these, it must stop the current transfer:
+
+  * *MGMT_ERR_EBADSTATE* - Possible reasons - device reboot occurred in the middle of the transfer.
+  * *MGMT_ERR_EINVAL* - The requested data structure cannot be decoded or incorrect information in the request was detected.
+  * *MGMT_ERR_ENOMEM* - Not enough space to store the image.
+  * *MGMT_ERR_ENOTSUP* - Unsupported command.
+
+  The Following errors are possibly recoverable:
```
Comment (richabp) [inline doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_smp.rst:382]:
```suggestion
  The following errors are possibly recoverable:
```

**Review by richabp — DISMISSED** [review]

**Review by richabp — APPROVED** [review]

**Review by SylwesterKonczyk — APPROVED** [review]

**Pull request merged**